src/Ng/Jobs/CatalogUtility.cs

```diff
@@ -79,9 +79,18 @@ public static HttpClient CreateHttpClient(bool verbose)
             var result = new SortedList<DateTime, IList<PackageDetails>>();
 
             XElement feed;
-            using (var stream = await client.GetStreamAsync(uri))
+            try
             {
-                feed = XElement.Load(stream);
+                using (var stream = await client.GetStreamAsync(uri))
+                {
+                    feed = XElement.Load(stream);
+                }
+            }
+            //if the client gets disposed because the timeout a task cancelled exception will happen
```

Comment:
Updated.

The above comment is a very good observation, with pros and cons that you mentioned.  In addition in the future we would probably like to integrate the exceptions with an alerting system as ICM. An  exception like this is not really actionable. Also having the transient exception concept we can implement retry patterns. In my opinion, would be good to avoid process termination if recovery is possible.